apiserver/facades/client/client/status.go

```diff
@@ -583,26 +575,26 @@ func (c *Client) modelStatus(ctx context.Context) (params.ModelStatusInfo, error
 	currentVersion := modelInfo.AgentVersion
 	info.Version = currentVersion.String()
 
-	m, err := c.stateAccessor.Model()
-	if err != nil {
-		return info, errors.Annotate(err, "cannot get model")
-	}
+	// // TODO: replace here once we implement the latest agent version in Dqlite.
+	// m, err := c.stateAccessor.Model()
+	// if err != nil {
+	// 	return info, errors.Annotate(err, "cannot get model")
+	// }
 
-	latestVersion := m.LatestToolsVersion()
-	if currentVersion.Compare(latestVersion) < 0 {
-		info.AvailableVersion = latestVersion.String()
-	}
+	// latestVersion := m.LatestToolsVersion()
+	// if currentVersion.Compare(latestVersion) < 0 {
+	// 	info.AvailableVersion = latestVersion.String()
+	// }
 
-	aStatus, err := m.Status()
+	aStatus, err := c.modelInfoService.Status(ctx)
 	if err != nil {
```

Comment:
You need to handle the model not found error here  and turn it into a not found error to keep satisfying the contract of the facade.

Reply:
Adding a test to assert that not found is returned as well would be good.